buildchain/targets/package.py

```diff
@@ -0,0 +1,290 @@
+        mkdir = Mkdir(directory=self.rootdir).task
+        task.update({
+            'name': self.MKDIR_TASK_NAME,
+            'doc': 'Create directory for {}.'.format(self.name),
+            'title': mkdir['title'],
+            'actions': mkdir['actions'],
+            'uptodate': mkdir['uptodate'],
+            'targets': mkdir['targets'],
+        })
+        return task
+
+    def generate_meta(self) -> types.TaskDict:
+        """Generate the .meta file for the package."""
+        task = self.basic_task
+        task.update({
+            'name': 'pkg_rpmspec',
+            'actions': [self._rpmspec],
+            'doc': 'Generate {}.meta'.format(self.name),
+            'title': lambda task: utils.title_with_target1('RPMSPEC', task),
+            'targets': [self.meta],
+        })
+        task['file_dep'].extend([self.spec, self.builder.destination])
+        task['task_dep'].append('{}:{}'.format(self.basename,
+                                               self.MKDIR_TASK_NAME))
+        return task
+
+    def get_source_files(self) -> types.TaskDict:
+        """Download the source files to build the package."""
+        targets = [self.srcdir]
+        targets.extend(self.sources)
+        actions = Mkdir(directory=self.srcdir).task['actions']
+        actions.append(self._download_sources)
+        task = self.basic_task
+        task.update({
+            'name': 'pkg_get_source',
+            'actions': actions,
+            'doc': 'Download source files for {}.'.format(self.name),
+            'title': lambda task: utils.title_with_target1('GET_SRC', task),
+            'targets': targets,
+        })
+        task['file_dep'].append(self.meta)
+        task['task_dep'].append('{}:{}'.format(self.basename,
+                                               self.MKDIR_TASK_NAME))
+        return task
+
+    def build_srpm(self) -> types.TaskDict:
+        """Build the SRPM for the package."""
+        task = self.basic_task
+        task.update({
+            'name': 'pkg_srpm',
+            'actions': [self._buildsrpm_cmd()],
+            'doc': 'Build {}'.format(self.srpm.name),
+            'title': lambda task: utils.title_with_target1('BUILD SRPM', task),
+            'targets': [self.srpm],
+            # Prevent Docker from polluting our output.
+            'verbosity': 0,
+        })
+        task['file_dep'].extend([self.spec, self.builder.destination])
+        task['file_dep'].extend(self.sources)
+        task['task_dep'].append('{}:{}'.format(self.basename,
+                                               self.MKDIR_TASK_NAME))
+        return task
+
+    def _rpmspec(self) -> None:
+        """Run the `rpmspec` command inside a container and save the output."""
+        container_spec = '/rpmbuild/SPECS/{}'.format(self.spec.name)
+        mount_string = 'type=bind,source={src},destination={dst},ro'.format(
+            src=self.spec, dst=container_spec
+        )
+        rpmspec_cmd = [
```

Comment:
Would it make sense to use the Docker Python API instead of always shell'ing out to Docker CLI?

Reply:
It could yeah.

My goal here was to:
- port the Makefile
- keep the build dependencies to a minimum (ideally only `doit`)

So I kept the CLI approach. But it may be worth/easier to use the Docker Python API.
If we want to go that way, i'll create a dedicated issue :-)

Reply:
I created an issue to keep track of that suggestion.